### PR TITLE
feat(create): variants config RTA handling

### PR DESCRIPTION
**Pull request by ytpo-lyne**
Enhancements for `@sap-ux/create add variants-config` command.

Variants-Config:
- uses yaml passed by cli or defaults to `ui5.yaml` (no longer loops over `ui5-local.yaml` or `ui5-mock.yaml`)
- introduced `--config` flag with used yaml to RTA script
- improved error handling when script already exists (checks if script is up to date)
- distinction between `fiori run` and `ui5 serve` based on used middleware

Additions:
- fix to read yaml from memory for `project-access`
- readMe update for `backend-proxy-middleware`

### CODE REVIEW

File: packages/app-config-writer/test/unit/variants-config/generateVariantsConfig.test.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ describe('generateVariantsConfig', () => {
     test('adding variants configuration to a non existing project', async () => {
         const basePath = join(__dirname, '../../fixtures/a-folder-that-does-not-exist');
         await expect(generateVariantsConfig(basePath, 'hugo.yaml', new ToolsLogger())).rejects.toThrowError(
-            `Script 'start-variants-management' cannot be written to package.json. File 'package.json' not found at ${basePath}`
+            `File 'hugo.yaml' not found in project '${basePath}'`
```
Comment (heimwege) [inline packages/app-config-writer/test/unit/variants-config/generateVariantsConfig.test.ts:21]: nit pick: why is the "Script 'start-variants-management' cannot be written to package.json." part gone?

File: packages/app-config-writer/src/variants-config/package-json.ts [hunk 2]
```diff
@@ -43,13 +40,26 @@ export async function addVariantsManagementScript(
...
+    const url = await getRTAUrl(basePath, getUI5UrlParameters(urlParameters), ui5YamlFileName);
+    const serveCommand = await getRTAServe(basePath, ui5YamlFileName, fs);
 
     if (!url) {
         throw new Error(`${ERROR_MSG} No RTA editor specified in ui5.yaml.`);
     }
 
-    packageJson.scripts['start-variants-management'] = `fiori run --open "${url}"`;
+    const startVariantsManagementScriptOld = packageJson.scripts['start-variants-management'] ?? undefined;
+    const startVariantsManagementScriptNew = `${serveCommand} --config ./${basename(ui5YamlFileName)} --open "${url}"`;
```
Comment (heimwege) [inline packages/app-config-writer/src/variants-config/package-json.ts:51]: @tobiasqueck @CG67 do we want to have --config mentioned in the script at all times or should we omit the default `ui5.yaml`).I'm indifferent. Opinions?

**Review by heimwege — COMMENTED** [review]
see comments

File: packages/app-config-writer/test/unit/variants-config/generateVariantsConfig.test.ts (hunk 1 above)
Comment (ytpo-lyne) [reply packages/app-config-writer/test/unit/variants-config/generateVariantsConfig.test.ts:21]: I [switched the methods](https://github.com/SAP/open-ux-tools/pull/2525/files#diff-9beb13032ed12a024e5c1dcfa70bdbbc0199fc10307b86633630b110c3a22227R27) to first update the yaml file accordingly. Therefore, the [error for a non-existing file](https://github.com/SAP/open-ux-tools/pull/2525/files#diff-40eac4eaa926a6b933a36d284af6c388f8f2668744420dca457fa61451f605f5R67) occurs at an earlier stage and not while handling the script insertion anymore.

File: .changeset/cold-trees-admire.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+'@sap-ux/app-config-writer': patch
+'@sap-ux/project-access': patch
+'@sap-ux/create': patch
+---
+
+Enhance RTA handling for varinats-config command
```
Comment (heimwege) [inline .changeset/cold-trees-admire.md:7]: typo

Comment (ytpo-lyne) [reply .changeset/cold-trees-admire.md]: Done 6818a0a817a528fc5aa9eb238f16eaf09d719bd2

**Review by heimwege — DISMISSED** [review]
changeset ok
changes are straightforward
excellent test coverage

File: packages/app-config-writer/src/variants-config/package-json.ts (hunk 2 above)
Comment (Klaus-Keller) [reply packages/app-config-writer/src/variants-config/package-json.ts:51]: I'd omit `--config` if it is using `ui5.yaml`.

Comment (ytpo-lyne) [reply packages/app-config-writer/src/variants-config/package-json.ts]: Done in 2dadf466926177a6fbf6ad132909f6c84ba17a1d

**Review by Klaus-Keller — APPROVED** [review]
Thanks @ytpo-lyne!
* changes for adding RTA support look goo
* changeset exists
* test coverage is great
* did a local test, no issues found

Approved from my side.

**Review by heimwege — APPROVED** [review]

**Pull request merged**